qiskit/circuit/quantumcircuit.py

```diff
@@ -586,6 +586,9 @@ def compose(self, other, qubits=None, clbits=None, front=False, inplace=False):
             dest = self.copy()
 
         if not isinstance(other, QuantumCircuit):
+            if isinstance(other, Instruction):
+                dest._update_parameter_table(other)
+
```

Comment:
This shouldn't be needed for `Instruction`s, `.append` and `.data.insert` should already take care of updating the `ParameterTable`.

Reply:
Ah, nice, thanks!